### PR TITLE
Add Zalm

### PR DESCRIPTION
**Pull request by tmagicturtle**

### CODE REVIEW

File: data/mods/ssb/statuses.ts [hunk 1]
```diff
@@ -607,6 +607,19 @@ export const BattleStatuses: {[k: string]: ModdedPureEffectData} = {
...
+	zalm: {
+		noCopy: true,
+		onStart() {
+			this.add(`c|${getName('Zalm')}|<(:O)000>`);
+		},
+		onSwitchOut(pokemon) {
+			this.add(`c|${getName('Zalm')}|Run for the hills!`);
+		},
+		onFaint(pokemon) {
+			const name = pokemon.side.foe.name;
```
Comment (ghost) [inline data/mods/ssb/statuses.ts]: delete

File: data/mods/ssb/moves.ts [hunk 2]
```diff
@@ -1656,6 +1656,36 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		flags: {},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onAfterHit(target, source) {
+			if (source.hp) {
+				const item = target.takeItem();
+				if (item) {
+					this.add('-enditem', source, target.item, '[of] ' + target, '[eat]', '[silent]'); // Silent otherwise it says that the target ate the item
+					this.heal(Math.floor(source.baseMaxhp * 0.5));
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts]:
```suggestion
					this.heal(source.baseMaxhp / 2);
```

File: data/mods/ssb/moves.ts [hunk 3]
```diff
@@ -1656,6 +1656,36 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		flags: {},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onAfterHit(target, source) {
+			if (source.hp) {
+				const item = target.takeItem();
+				if (item) {
+					this.add('-enditem', source, target.item, '[of] ' + target, '[eat]', '[silent]'); // Silent otherwise it says that the target ate the item
+					this.heal(source.baseMaxhp / 2);
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts]: Did you test this code? i'm pretty sure this code throws an error.

Comment (tmagicturtle) [reply data/mods/ssb/moves.ts]: It doesn't, but it did accidentally heal the target rather than the user.

File: data/mods/ssb/moves.ts [hunk 4]
```diff
@@ -1656,6 +1656,36 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		priority: 0,
+		flags: {},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onAfterHit(target, source) {
+			if (source.hp) {
+				const item = target.takeItem();
+				if (item) {
+					this.add('-enditem', source, target.item, '[of] ' + target, '[eat]', '[silent]'); // Silent otherwise it says that the target ate the item
```
Comment (ghost) [inline data/mods/ssb/moves.ts]:
```suggestion
					this.add('-enditem', source, target.item, '[of] ' + target, '[eat]');
```
fwiw, that IS what is happening lol

Comment (tmagicturtle) [reply data/mods/ssb/moves.ts]: No, the Weedle eats the target's item, not the target

Comment (ghost) [reply data/mods/ssb/moves.ts]: OH

Comment (ghost) [reply data/mods/ssb/moves.ts]: honestly tbh I feel like you can look @ pluck and use that code.

**Pull request merged**